### ISSUE TITLE
[wg/browser-tools-testing] Browser Tools and Testing WG (WebDriver) rechartering

### ISSUE THREAD

**Issue by sideshowbarker** [w3c/strategy#438]
New charter proposal, reviewers please take note.

# Charter Review

Charter: https://w3c.github.io/charter-drafts/2024/btt-wg.html

What kind of charter is this?
 - [X] Existing WG recharter; [diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2021%2F08%2Fbbtt-wg-charter.html&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2024%2Fbtt-wg.html)

For responses or questions about this draft charter, please add comments to this issue.

**Comment by ruoxiran** [comment]
no comments or requests from APA.

**Comment by plehegar** [comment]
(pinging @w3c/w3c-group-52497-chairs in case this is of interest)

**Comment by himorin** [comment]
- two section numbers of the Process in "About this charter" section seem old one
- history table needs to have recent extension and this charter?

**Comment by sideshowbarker** [comment]
> * two section numbers of the Process in "About this charter" section seem old one

Thanks for catching that — fixed now

> * history table needs to have recent extension and this charter?

OK — added those

> * not sure why (almost?) every 'should' are replaced with 'will' in success criteria section, but for me one for 'security and privacy implications' seems to be 'should' per recent requests around...

I replaced those because the use of _“should”_ throughout the charter template seems weird to me. It seems like the core purpose of the charter is for us to define normative “must” _requirements_ for the group (the group _will_ do these things) — not general “nice to have if you feel like doing them” things (the group _should_ do these things).

But I don’t feel particularly strongly about it at all. I’m very happy to change them all back to _“should”_.

Shall I go ahead and do that?

**Comment by plehegar** [comment]
Advance notice: https://lists.w3.org/Archives/Member/w3c-ac-members/2023OctDec/0037.html

**Comment by plehegar** [comment]
No comment from PING

**Comment by plehegar** [comment]
@ruoxiran @matatk @JaninaSajka 

AT Driver was added to the charter:
[[
[AT Driver](https://w3c.github.io/at-driver/)

AT Driver defines a protocol for introspection and remote control of assistive technology (AT) such as screen readers, using a bidirectional communication channel.
]]


If this is a concern for APA, please let know. Keep in mind that this would only be enabled in testing environment.

**Comment by ruoxiran** [comment]
> @ruoxiran @matatk @JaninaSajka
> 
> AT Driver was added to the charter: [[ [AT Driver](https://w3c.github.io/at-driver/)
> 
> AT Driver defines a protocol for introspection and remote control of assistive technology (AT) such as screen readers, using a bidirectional communication channel. ]]
> 
> If this is a concern for APA, please let know. Keep in mind that this would only be enabled in testing environment.

thanks, PLH, we will review it this week.

**Comment by plehegar** [comment]
Deliverables are missing Draft state, Expected completion, Adopted Draft, Exclusion Draft, and Exclusion Draft Charter. (see [charter template](https://w3c.github.io/charter-drafts/charter-template.html) and [webapps charter example](https://www.w3.org/2024/01/webappswg-charter-2024.html#deliverablesfull)).


Is there WHATWG coordination needed related to [Test Utils](https://testutils.spec.whatwg.org/)?

Shouldn't we list WPT in coordination as well?

In charter history, "AT Driver deliverable added." is listed as added in 2021. But isn't it 2024?

**Comment by ruoxiran** [comment]
> > @ruoxiran @matatk @JaninaSajka
> > AT Driver was added to the charter: [[ [AT Driver](https://w3c.github.io/at-driver/)
> > AT Driver defines a protocol for introspection and remote control of assistive technology (AT) such as screen readers, using a bidirectional communication channel. ]]
> > If this is a concern for APA, please let know. Keep in mind that this would only be enabled in testing environment.
> 
> thanks, PLH, we will review it this week.

Sorry for the delay, we need one more week to follow up with AT vendors. See minutes at: https://www.w3.org/2024/03/06-apa-minutes.html#t04

**Comment by svgeesus** [comment]
The current charter has a mix of "living standard" at the top and "going to rec" at the bottom.

**Comment by sideshowbarker** [comment]
> The current charter has a mix of "living standard" at the top and "going to rec" at the bottom.

Thanks for catching that — I’ve now removed the _“In order to advance to Proposed Recommendation”_ paragraph from the **Success Criteria** section. Lemme know if I missed anything else.

**Comment by sideshowbarker** [comment]
> Does the Group want to produce a Recommendation or produce/update Candidate Recommendation Snapshots (Living-REC vs Living-CR)?

For the record here, the answer to that is, the group does not intend to produce any RECs but instead plans to publish CR snapshots (I was reminded yesterday that the group had already made a decision about that during a TPAC discussion quite a while ago)

**Comment by sideshowbarker** [comment]
> Deliverables are missing Draft state, Expected completion, Adopted Draft, Exclusion Draft, and Exclusion Draft Charter. (see [charter template](https://w3c.github.io/charter-drafts/charter-template.html) and [webapps charter example](https://www.w3.org/2024/01/webappswg-charter-2024.html#deliverablesfull)).

So now that we have confirmation that the group in fact does not intend to produce Recommendation-track deliverables, do we need to include the _“Draft state, Adopted Draft, Exclusion Draft, and Exclusion Draft Charter”_ info in the charter?

I ask that because in the Process doc at https://www.w3.org/2023/Process-20231103/#WGCharter I see:

> _For every Recommendation Track deliverable that continues work on [technical report](https://www.w3.org/2023/Process-20231103/#technical-report) published under any other Charter (including a predecessor group of the same name), for which there is at least an existing [First Public Working Draft](https://www.w3.org/2023/Process-20231103/#fpwd) the description of that deliverable in the proposed charter of the adopting Working Group must provide the following information:_

But if the group only plans to take their deliverables to CR Snapshot, doesn’t that mean the deliverables are _not_ Recommendation Track?

It’s not completely clear to me at least that the Process document intends for CR Snapshots to be considered Recommendation-Track deliverables for the purposes of that requirement to include the _“Draft state, Adopted Draft, Exclusion Draft, and Exclusion Draft Charter”_ info in charters.

At https://www.w3.org/2023/Process-20231103/#w3c-recommendation-track I see this:

> In summary, the W3C Recommendation Track consists of:
> 1. Publication of the [First Public Working Draft](https://www.w3.org/2023/Process-20231103/#fpwd).
> 2. Publication of zero or more revised [Working Drafts](https://www.w3.org/2023/Process-20231103/#RecsWD).
> 3. Publication of one or more [Candidate Recommendations](https://www.w3.org/2023/Process-20231103/#RecsCR).
> 4. Publication of a [Proposed Recommendation](https://www.w3.org/2023/Process-20231103/#RecsPR).
> 5. Publication as a [W3C Recommendation](https://www.w3.org/2023/Process-20231103/#RecsW3C).

…with the implication (to me at least) that for a deliverable to be considered a Recommendation-Track deliverable, it needs to be intended to eventually complete _all_ those steps — not just the first 3.

And the intro paragraph to the https://www.w3.org/2023/Process-20231103/#rec-track section says, _“These technical reports undergo cycles of revision and review as they advance towards W3C Recommendation status”_ — but CR Snapshots are explicitly not intended to advance toward W3C Recommendation status.

Given all that, it would seem like the “must include _“Draft state, Adopted Draft, Exclusion Draft, and Exclusion Draft Charter”_ charter requirement isn’t intended to apply to CR-Snapshot-track deliverables.

**Comment by plehegar** [comment]
> So now that we have confirmation that the group in fact does not intend to produce Recommendation-track deliverables, do we need to include the _“Draft state, Adopted Draft, Exclusion Draft, and Exclusion Draft Charter”_ info in the charter?

You still need to include those information. It's relevant for the patent policy.

> > In summary, the W3C Recommendation Track consists of:
> > 
> > 1. Publication of the [First Public Working Draft](https://www.w3.org/2023/Process-20231103/#fpwd).
> > 2. Publication of zero or more revised [Working Drafts](https://www.w3.org/2023/Process-20231103/#RecsWD).
> > 3. Publication of one or more [Candidate Recommendations](https://www.w3.org/2023/Process-20231103/#RecsCR).
> > 4. Publication of a [Proposed Recommendation](https://www.w3.org/2023/Process-20231103/#RecsPR).
> > 5. Publication as a [W3C Recommendation](https://www.w3.org/2023/Process-20231103/#RecsW3C).
> 
> …with the implication (to me at least) that for a deliverable to be considered a Recommendation-Track deliverable, it needs to be intended to eventually complete _all_ those steps — not just the first 3.

I guess we could replace /consists/contains/ to make it clearer that REC-track does not mean systematically means that the document will reach the REC maturity stage. cc @frivoal just in case.

**Comment by svgeesus** [comment]
> But if the group only plans to take their deliverables to CR Snapshot, doesn’t that mean the deliverables are not Recommendation Track?

Well they aren't Notes and they are not Registries. So they are Rec-track (consider the name historical, standards-track would be better)

**Comment by matatk** [comment]
APA WG is happy for this to proceed.

We are still engaging with external community stakeholders, but we are mindful that we did not spot any showstoppers ourselves, and we will have the opportunity to provide feedback on the AT Driver spec as it progresses (and we'll help anyone in the wider accessibility community who wants to, to provide feedback as it comes in). Thanks for your patience!

cc @ruoxiran @JaninaSajka

**Comment by simoneonofri** [comment]
From a security perspective, they should include in the Success Criteria a precise structure of the "Security Considerations" Section, which has already been identified any way in the different issues of the deliverables.

> The security considerations section must include a comprehensive threat model with threats, attacks, mitigations, and residual risks.

This section's structure is indicated by the [Security and Privacy Questionnaire](https://www.w3.org/TR/security-privacy-questionnaire/#considerations) and the referenced [RFC3552—Guidelines for Writing RFC Text on Security Considerations](https://www.rfc-editor.org/rfc/rfc3552).

In this specific case, it should be interesting to consider potential abuse cases of the APIs/Driver both from user perspectives and use by bots for scraping.

**Comment by dontcallmedom** [comment]
[CfP announced on Jul 9](https://lists.w3.org/Archives/Public/public-new-work/2024Jul/0002.html)
